### ISSUE TITLE
Support geohash queries

### ISSUE THREAD

**Issue by MatthewJA** [TerriaJS/terriajs#4785]
Feature request to support spatial queries by [geohash](http://en.wikipedia.org/wiki/Geohash). This would be really useful for projects which name polygons or points by their location.

**Comment by rowanwins** [comment]
Hey @MatthewJA 

Thanks for the suggestion, can you perhaps just expand how you'd envisage this working?

Thanks,
Rowan

**Comment by MatthewJA** [comment]
Hi @rowanwins, sure thing. The search box could take geohash queries:
![image](https://user-images.githubusercontent.com/2681312/94227691-712c3800-ff3e-11ea-9c1c-b1fb629d1d41.png)
which would resolve as locations:
![image](https://user-images.githubusercontent.com/2681312/94227724-91f48d80-ff3e-11ea-821a-3045a287cf54.png)

Here's an image of a dataset that makes use of geohashes already as identifiers:
![image](https://user-images.githubusercontent.com/2681312/94227777-b18bb600-ff3e-11ea-9d2e-5bbecc66216d.png)

The current workflow for using geohashes is to first convert them with a service like [http://geohash.co/](geohash.co) into lat/lon and then search for that instead, which resolves to a coordinate location.

**Comment by rowanwins** [comment]
Yeah cool, the search box was the easy (ier) thing that came to mind. Thanks for the clarification.

**Comment by KeyboardSounds** [comment]
We should be able to support this relatively easily using the new [Web Feature Service search provider](https://github.com/TerriaJS/terriajs/blob/next/lib/Models/WebFeatureServiceSearchProvider.ts) since there's a WFS version of this layer.

**Comment by rowanwins** [comment]
Or pull in an npm package like [ngeohash](https://www.npmjs.com/package/ngeohash) or [latlon-geohash ](https://www.npmjs.com/package/latlon-geohash) which does the decoding clientside

**Comment by KeyboardSounds** [comment]
@rowanwins I like your idea better, because it's much more generic. The WFS search provider approach is layer-specific, but decoding a geohash isn't.
